test/replication-luatest/gh_6033_box_promote_demote_test.lua

```diff
@@ -0,0 +1,501 @@
+local luatest = require('luatest')
+local server = require('test.luatest_helpers.server')
+local cluster = require('test.luatest_helpers.cluster')
+local g_nspaces = luatest.group('gh-6033-box-promote-demote')
+local g_unconfigured = luatest.group('gh-6033-box-promote-demote-unconfigured')
+local g_spaces = luatest.group('gh-6033-box-promote-demote-with-spaces',
+    {{engine = 'memtx'}, {engine = 'vinyl'}})
+
+local promote_start = function(server)
+    return server:exec(function()
+        local f = require('fiber').new(box.ctl.promote)
+        f:set_joinable(true)
+        return f:id()
+    end)
+end
+
+local demote_start = function(server)
+    return server:exec(function()
+        local f = require('fiber').new(box.ctl.demote)
+        f:set_joinable(true)
+        return f:id()
+    end)
+end
+
+local fiber_join = function(server, fiber)
+    return server:exec(function(fiber)
+        return require('fiber').find(fiber):join()
+    end, {fiber})
+end
+
+local wal_delay_start = function(server,  countdown)
+    if countdown == nil then
+        server:exec(function()
+            box.error.injection.set('ERRINJ_WAL_DELAY', true)
+        end)
+    else
+        server:exec(function(countdown)
+            box.error.injection.set('ERRINJ_WAL_DELAY_COUNTDOWN', countdown)
+        end, {countdown})
+    end
+end
+
+local wal_delay_end = function(server)
+    server:exec(function()
+        box.error.injection.set('ERRINJ_WAL_DELAY', false)
+    end)
+end
+
+local cluster_init = function(g)
+    g.cluster = cluster:new({})
+
+    local engine = 'memtx'
+    if g.params ~= nil and g.params.engine ~= nil then
+        engine = g.params.engine
+    end
+
+    g.box_cfg = {
+        election_mode = 'off',
+        election_timeout = 0.5,
+        read_only = false,
+        replication_timeout = 0.1,
+        replication_synchro_timeout = 5,
+        replication_synchro_quorum = 1,
+        replication = {
+            server.build_instance_uri('server_1'),
+            server.build_instance_uri('server_2'),
+        },
+    }
+
+    g.server_1 = g.cluster:build_and_add_server(
+        {alias = 'server_1', engine = engine, box_cfg = g.box_cfg})
+    g.server_2 = g.cluster:build_and_add_server(
+        {alias = 'server_2', engine = engine, box_cfg = g.box_cfg})
+    g.cluster:start()
+
+    g.current_test = 1
```

Comment:
Why do you need that?

Reply:
Whoops, left it behind after deleting debug prints.